edi_oca/models/edi_configuration.py

```diff
@@ -0,0 +1,189 @@
+
+        :returns: dict -- evaluation context given to safe_eval
+        """
+        ctx = {
+            "uid": self.env.uid,
+            "user": self.env.user,
+            "DotDict": DotDict,
+            "conf": self,
+        }
+        ctx.update(self._time_utils())
+        return ctx
+
+    def _evaluate_code_snippet(self, snippet, **render_values):
+        if not self._code_snippet_valued(snippet):
+            return {}
+        eval_ctx = dict(render_values, **self._get_code_snippet_eval_context())
+        safe_eval.safe_eval(snippet, eval_ctx, mode="exec", nocopy=True)
+        result = eval_ctx.get("result", {})
+        if not isinstance(result, dict):
+            return {}
+        return result
+
+    def edi_exec_snippet_before_do(self, record, **kwargs):
+        self.ensure_one()
+        todo = kwargs.get("todo", False)
+        vals = {
+            "todo": todo,
+        }
+        # Execute snippet before do
+        vals_before_do = self._evaluate_code_snippet(
+            self.snippet_before_do, record=record
+        )
+
+        # Prepare data
+        vals.update(
+            {
+                "todo": vals_before_do.get("todo", False),
+                "snippet_do_vars": vals_before_do.get("snippet_do_vars", False),
+                "event_only": vals_before_do.get("event_only", False),
+                "tracked_fields": vals_before_do.get("tracked_fields", False),
+                # Etc ...
+            }
+        )
+        # Update EDIAutoInfo if "snippet_do_vars" in vals
+        # TODO: Add check value in snippet_do_vars
+        if vals.get("snippet_do_vars", False) and todo:
+            for rec in todo:
+                rec.reinitialize(vals["snippet_do_vars"])
+            vals.update({"todo": todo})
+        return vals
+
+    def edi_exec_snippet_do(self, record, **kwargs):
+        self.ensure_one()
+        old_value = kwargs.get("old_vals", {}).get(record.id, False)
+        new_value = kwargs.get("vals", {}).get(record.id, False)
+        vals = {
+            "record": record,
+            "operation": kwargs.get("operation", False),
+            "edi_action": kwargs.get("edi_action", False),
+            "old_value": old_value,
+            "vals": new_value,
+        }
+        if self.snippet_before_do:
+            before_do_vals = self.edi_exec_snippet_before_do(record, **kwargs)
+            vals.update(before_do_vals)
+        if vals.get("todo", False) and getattr(self, "_edi_auto_handle", False):
+            record._edi_auto_handle(vals["todo"])
+        self._evaluate_code_snippet(self.snippet_do, **vals)
+
+    @api.model
+    def edi_get_conf(self, trigger, backend=None):
+        domain = [("trigger", "=", trigger)]
+        if backend:
+            domain.append(("backend_id", "=", backend.id))
+        else:
+            domain.append(("backend_id", "=", self.type_id.backend_id.id))
+        return self.search(domain)
```

Comment:
Missing `active` ?
Could we make this search smatter with `model` and `type` (optional)?